worker/caasunitprovisioner/deployment_worker.go

```diff
@@ -138,20 +130,11 @@ func (w *deploymentWorker) loop() error {
 		if err != nil {
 			return errors.Annotate(err, "cannot parse pod spec")
 		}
-		// When Juju manages the units, we don't ask the
-		// substrate to create any itself.
-		if w.jujuManagedUnits {
-			numUnits = 0
-		}
 		err = w.broker.EnsureService(w.application, spec, numUnits, appConfig)
```

Comment:
The EnsureService method has code that only creates a Deployment Controller if num units > 0.
That code can be simplified now. And there's also a check for num units >=0 which should now be just > 0